cookbook/case_studies/house_price_prediction/house_price_predictor.py

```diff
@@ -0,0 +1,232 @@
+"""
+Predicting House Price in a Region Using an XGBoost Model and Flytekit (Python)
+-------------------------------------------------------------------------------
+
+"""
+
+# %%
+# Install the following three libraries before running the model (locally):
+# .. code-block:: python
+#
+#       pip install scikit-learn
+#       pip install joblib
+#       pip install xgboost
+
+# %%
+# Step 1: Importing the Libraries
+# -------------------------------
+import typing
+
+import joblib
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import train_test_split
+from xgboost import XGBRegressor
+from flytekit import Resources, dynamic, task, workflow
+from flytekit.types.file import FlyteFile
+
+# %%
+# Step 2: Initializing the Variables
+# ----------------------------------
+NUM_HOUSES_PER_LOCATION = 1000
+COLUMNS = [
+    "PRICE",
+    "YEAR_BUILT",
+    "SQUARE_FEET",
+    "NUM_BEDROOMS",
+    "NUM_BATHROOMS",
+    "LOT_ACRES",
+    "GARAGE_SPACES",
+]
+MAX_YEAR = 2021
+SPLIT_RATIOS = [0.6, 0.3, 0.1]
+
+# %%
+# Step 3: Defining the Data Generation Functions
+# ----------------------------------------------
+#
+# House price generation function:
+def gen_price(house) -> int:
+    _base_price = int(house["SQUARE_FEET"] * 150)
+    _price = int(
+        _base_price
+        + (10000 * house["NUM_BEDROOMS"])
+        + (15000 * house["NUM_BATHROOMS"])
+        + (15000 * house["LOT_ACRES"])
+        + (15000 * house["GARAGE_SPACES"])
+        - (5000 * (MAX_YEAR - house["YEAR_BUILT"]))
+    )
+    return _price
+
+
+# %%
+# Dataframe with All the House Details:
+def gen_houses(num_houses) -> pd.DataFrame:
+    _house_list = []
+    for _ in range(num_houses):
+        _house = {
+            "SQUARE_FEET": int(np.random.normal(3000, 750)),
+            "NUM_BEDROOMS": np.random.randint(2, 7),
+            "NUM_BATHROOMS": np.random.randint(2, 7) / 2,
+            "LOT_ACRES": round(np.random.normal(1.0, 0.25), 2),
+            "GARAGE_SPACES": np.random.randint(0, 4),
+            "YEAR_BUILT": min(MAX_YEAR, int(np.random.normal(1995, 10))),
+        }
+        _price = gen_price(_house)
+        _house_list.append(
+            [
+                _price,
+                _house["YEAR_BUILT"],
+                _house["SQUARE_FEET"],
+                _house["NUM_BEDROOMS"],
+                _house["NUM_BATHROOMS"],
+                _house["LOT_ACRES"],
+                _house["GARAGE_SPACES"],
+            ]
+        )
+    _df = pd.DataFrame(
+        _house_list,
+        columns=COLUMNS,
+    )
+    return _df
+
+
+# %%
+# Splitting the Data into Train, Val, and Test Datasets:
+def split_data(
```

Comment:
can we use a named output?
NamedTuple?

Reply:
Observing an issue concerning "NamedTuple" (`Error:RestrictedTypeError(f"Transformer for type {self.python_type} is restricted currently"`)). Looking into it.